catalyst-gateway/bin/src/db/event/config/mod.rs

```diff
@@ -0,0 +1,65 @@
+//! Configuration query
+
+use jsonschema::BasicOutput;
+use key::ConfigKey;
+use serde_json::Value;
+use tracing::error;
+
+use crate::db::event::EventDB;
+
+pub(crate) mod key;
+
+/// Configuration struct
+pub(crate) struct Config {}
+
+/// SQL get configuration.
+const GET_CONFIG: &str = include_str!("sql/get.sql");
+/// SQL update if exist or else insert configuration.
+const UPSERT_CONFIG: &str = include_str!("sql/upsert.sql");
+
+impl Config {
+    /// Retrieve configuration based on the given `ConfigKey`.
+    ///
+    /// # Returns
+    ///
+    /// - A JSON value of the configuration, if not found, returns the default value.
+    /// - Error if the query fails.
+    pub(crate) async fn get(id: ConfigKey) -> anyhow::Result<Value> {
+        let (id1, id2, id3) = id.to_id();
+        let rows = EventDB::query(GET_CONFIG, &[&id1, &id2, &id3]).await?;
+
+        if let Some(row) = rows.first() {
+            let value: Value = row.get(0);
+            match id.validate(&value) {
+                BasicOutput::Valid(_) => Ok(value),
+                BasicOutput::Invalid(errors) => {
+                    // This should not happen; expecting the schema to be valid
+                    error!("Validate schema failed: {:?}", errors);
```

Comment:
```suggestion
                    error!(id=%id, errors=?errors, "Get Config, Schema validation failed.  Defaulting.");
```
Add the config key here (its name derived from the enum is sufficient).  
Otherwise when the error gets thrown, we won't know WHAT was failing when there are multiple configs we can read.
I would also put the errors and ConfigKey name in as data values, not printed because its better for log parsing later.
This requires `id` to implement `Display` errors are displayed using debug.